server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -81,6 +84,11 @@ public int getCurrentQueueSize() {
         return getQueue().size();
     }
 
+    @Override
+    protected void beforeExecute(Thread t, Runnable r) {
+        ongoingTasks.put(r, System.nanoTime());
```

Comment:
Do you need the accuracy of `System.nanoTime()` or would you be fine with the more performant but less accurate thread cached time (`absoluteTimeInMillis()`)?

Reply:
The wrapper (`TimedRunnable`) that records the total execution time of the task is using `System.nanoTime()`, which is why I've also chosen the same time source to avoid any surprises. I mean we could reuse the recorded value there by unwrapping and exposing the recorded start time, not sure if that just complicates things, but it's an option.

Reply:
Actually let me try...

Reply:
I've pushed https://github.com/elastic/elasticsearch/pull/97661/commits/35529f837562736c94252e91a1e936bf5180a80f.

Reply:
I've reverted https://github.com/elastic/elasticsearch/commit/35529f837562736c94252e91a1e936bf5180a80f and added a comment in https://github.com/elastic/elasticsearch/pull/97661/commits/d37ba5d950574ed1409ed5446cfcf985b77ed305. I think reusing the timestamp of the wrapper is not correct. That timestamp starts when the task in submitted to the executor. We want the timestamp of when it actually starts execution which could be (much) later, therefore we need to capture the time again. And I'd rather use the same source.

Reply:
Ah I had not realized that TimedRunnable is already using System.nanoTime(). Then LGTM